src/vec_array.rs

```diff
@@ -0,0 +1,22 @@
+use crate::{Encode, Decode, Error, Output, Input};
+use crate::alloc::vec::Vec;
+use core::convert::TryFrom;
+
+impl<T: Encode, L: typenum::Unsigned> Encode for vecarray::VecArray<T, L> {
+	fn encode_to<W: Output>(&self, dest: &mut W) {
+		for item in self.iter() {
+			item.encode_to(dest);
+		}
+	}
+}
+
+impl<T: Decode, L: typenum::Unsigned> Decode for vecarray::VecArray<T, L> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let mut r = Vec::new();
```

Comment:
wouldn't it be better to have with_capacity here ? it doesn't harm as this capacity is on the type itself, nobody can craft it with wrong input.